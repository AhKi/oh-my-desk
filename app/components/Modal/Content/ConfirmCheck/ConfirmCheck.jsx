import React from 'react';
import PropTypes from 'prop-types';
import './ConfirmCheck.scss';

const propTypes = {
  cancelText: PropTypes.string,
  confirmText: PropTypes.string,
  content: PropTypes.string,
  title: PropTypes.string,
  onConfirm: PropTypes.func,
  onModalClose: PropTypes.func,
};
const defaultProps = {
  cancelText: 'close',
  confirmText: 'ok',
  content: '',
  title: '',
  onConfirm() {},
  onModalClose() {},
};

class ConfirmCheck extends React.Component {
  constructor(props) {
    super(props);
    this.handleConfirm = this.handleConfirm.bind(this);
  }

  handleConfirm(arg) {
    this.props.onConfirm(arg);
    this.props.onModalClose();
  }

  render() {
    const {
      cancelText,
      confirmText,
      content,
      title,
      onModalClose,
    } = this.props;

    return (
      <div className="ConfirmCheck">
        <h3 className="ConfirmCheck__title">{title}</h3>
        <p className="ConfirmCheck__content">{content}</p>
        <button
          type="button"
          className="Btn Btn--primary Btn-middle"
          onClick={this.handleConfirm}
        >
          {confirmText}
        </button>
        <button
          type="button"
          className="Btn Btn--primary Btn-middle"
          onClick={onModalClose}
        >
          {cancelText}
        </button>
      </div>
    );
  }
}

ConfirmCheck.propTypes = propTypes;
ConfirmCheck.defaultProps = defaultProps;

export default ConfirmCheck;
