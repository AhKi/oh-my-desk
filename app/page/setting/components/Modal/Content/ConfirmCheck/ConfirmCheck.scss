.ConfirmCheck {
  width: 300px;
  padding: 25px;
  overflow: auto;
  text-align: center;
}


.ConfirmCheck__title {
  text-align: center;
  font-weight: 600;
  color: $color-middle-gray-blue;
}

.ConfirmCheck__content {
  font-weight: 400;
  font-size: 13px;
  color: $color-middle-gray-blue;
  display: inline-block;
  text-align: center;
}

.ConfirmCheck__btn {
  display: flex;
  justify-content: center;
}

.ConfirmCheck__btn-box {
  display: flex;
  justify-content: center;
}