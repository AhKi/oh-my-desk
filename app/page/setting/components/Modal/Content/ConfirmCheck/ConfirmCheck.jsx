import React from 'react';
import PropTypes from 'prop-types';
import './ConfirmCheck.scss';

const propTypes = {
  cancelText: PropTypes.string,
  confirmText: PropTypes.string,
  content: PropTypes.string,
  title: PropTypes.string,
  onConfirm: PropTypes.func,
  onModalClose: PropTypes.func,
};
const defaultProps = {
  cancelText: 'cancel',
  confirmText: 'ok',
  content: '',
  title: '',
  onConfirm() {},
  onModalClose() {},
};

class ConfirmCheck extends React.Component {
  constructor(props) {
    super(props);
    this.handleConfirm = this.handleConfirm.bind(this);
  }

  handleConfirm(arg) {
    this.props.onConfirm(arg);
    this.props.onModalClose();
  }

  render() {
    const {
      cancelText,
      confirmText,
      content,
      title,
      onModalClose,
    } = this.props;

    return (
      <div className="ConfirmCheck">
        <h6 className="ConfirmCheck__title space-2x"><strong>{title}</strong></h6>
        <span
          className="ConfirmCheck__content space-4x"
        >
          {content.split('\n').map((line, index) => (
            <span key={index}>{line}<br /></span> // eslint-disable-line react/no-array-index-key
          ))}
        </span>
        <div className="ConfirmCheck__btn-box">
          <button
            type="button"
            className="Btn Btn--gray Btn--sm"
            onClick={onModalClose}
          >
            {cancelText}
          </button>
          <button
            type="button"
            className="Btn Btn--primary Btn--sm ConfirmCheck__btn"
            onClick={this.handleConfirm}
          >
            {confirmText}
          </button>
        </div>
      </div>
    );
  }
}

ConfirmCheck.propTypes = propTypes;
ConfirmCheck.defaultProps = defaultProps;

export default ConfirmCheck;
