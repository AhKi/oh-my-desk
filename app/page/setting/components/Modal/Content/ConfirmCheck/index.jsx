import { connect } from 'react-redux';
import { modalClose } from 'setting/store/modal/actions';

import ConfirmCheck from './ConfirmCheck';

const mapDispatchToProps = {
  onModalClose: modalClose,
};

export default connect(null, mapDispatchToProps)(ConfirmCheck);
