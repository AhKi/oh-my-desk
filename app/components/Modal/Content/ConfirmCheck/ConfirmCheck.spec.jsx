import React from 'react';
import { shallow } from 'enzyme';

import ConfirmCheck from './ConfirmCheck';

describe('<ConfirmCheck />', () => {
  it('should match to snapshot when render default', () => {
    const wrapper = shallow(<ConfirmCheck />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should call onConfirm when call handleConfirm', () => {
    const onConfirm = jest.fn();
    const onModalClose = jest.fn();
    const wrapper = shallow(
      <ConfirmCheck
        onConfirm={onConfirm}
        onModalClose={onModalClose}
      />,
    );

    wrapper.instance().handleConfirm('mock-arg');

    expect(onConfirm).toHaveBeenCalledTimes(1);
    expect(onConfirm).toHaveBeenCalledWith('mock-arg');
    expect(onModalClose).toHaveBeenCalledWith();
  });
});
