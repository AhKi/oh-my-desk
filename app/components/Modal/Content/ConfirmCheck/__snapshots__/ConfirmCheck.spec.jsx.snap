// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmCheck /> should match to snapshot when render default 1`] = `
<div
  className="ConfirmCheck"
>
  <h3
    className="ConfirmCheck__title"
  />
  <p
    className="ConfirmCheck__content"
  />
  <button
    className="Btn Btn--primary Btn-middle"
    onClick={[Function]}
    type="button"
  >
    ok
  </button>
  <button
    className="Btn Btn--primary Btn-middle"
    onClick={[Function]}
    type="button"
  >
    close
  </button>
</div>
`;
